docs/adr/0014-on-chain-state-management.md

```diff
@@ -0,0 +1,162 @@
+
+## Status
+
+:hammer_and_wrench:
+
+## Context
+
+The Hydra head protocol requires a few on-chain interactions. Those are governed by on-chain smart validator scripts which ensures the correct execution of the Hydra head state-machine. Each transition of the state machine takes the form of a transaction on-chain. As part of the Hydra nodes, we have to observe the chain and react to transitions accordingly, while maintaining a state across all distributed Hydra nodes.
+
+Our current approach involves maintaining a so-called 'OnChainHeadState' sum-type containing aggregated information and capturing the state-machine's state. Beside, we identify transactions to be observed using traits of those transactions and extract for each informations relevant to the next transitions. 
+
+We've been bothered with maintaining an 'OnChainHeadState' ever since we went for this approach. Indeed, it is a partially redundant piece of information which overlaps with the core of the Hydra node which is managing the application logic (and also maintains a folded state). Plus, with the introduction of head identification and multi-head support, we are exploring ways to simplify and make less cumbersome our on-chain state management.
+
+## Decision
+
+A first observation we made is that we can actually observe all transactions involved in the head life-cycle by solely looking at those transactions and some credentials known of the node (i.e. a verification key). From each transaction, we want to be able to derive _some identifier_ that uniquely identifies a head in order to tell head apart. Before anything else, let's recap how we currently observe each transaction on-chain. 
+
+### Head Life-cycle Summary
+
+To ease notation, here below is a table which summarizes the name of each transition and the states they transition from and to. 
+
+| Name       | From    | To      |
+| ---        | ---     | ---     |
+| Init       | -       | Initial |
+| Commit     | Initial | Initial |
+| Abort      | Initial | -       |
+| CollectCom | Initial | Open    |
+| Close      | Open    | Closed  |
+| Fanout     | Closed  | -       |
+
+We assume that all nodes are configured with:
+
+- A Cardano verification key `vk_cardano` 
+- An ephemeral verification key `vk_ephemeral`
+- A `νHead`, `νInitial`, `νCommit` scripts (for which one can compute static hashes)
+
+#### Observing an 'Init' transaction
+
+A transaction qualifies as an `Init` if and only if:
+
+- It produces exactly one output to an address locked by the `νHead` script, with a datum `δHead` pointing at an `Initial` state. In particular, `δHead` contains a list of `n` verification key hashes identifying participating parties.
+
+- Our verification key `vk_ephemeral` is a member of the list of ephemeral keys stored in `δHead`.
+
+- For each key hash in `δHead`, it mints a participation token with a unique policy id and such that its asset name is an injective function of the corresponding key hash it's minted for. 
```

Comment:
This is essentially what we are aiming for (not the current status quo). Might be nice to link to the token ADR..14 (? 🙂) here now.